src/stores/DAOEntityStore.js

```diff
@@ -121,14 +121,36 @@ class DAOEntityStore<TEntity: { id: string }, TEntityMutator: {}> {
     this.entityItemsByID.get(id),
   );
 
-  getByQueryFilters = createTransformer(
-    (queryFilters: Array<QueryFilter>): Array<TEntity> =>
-      this.entityItemsByID
-        .values()
-        .filter((entityItem: TEntity): boolean =>
-          DAOApi.doesSatisfyToQueryFilters(entityItem, queryFilters),
-        ),
-  );
+  getByQueryOptions = createTransformer((queryOptions: QueryOptions): Array<
+    TEntity,
+  > => {
+    const { filters, orderBy } = queryOptions;
+    let entities = this.all;
+
+    if (filters) {
+      entities = entities.filter((entityItem: TEntity): boolean =>
+        DAOApi.doesSatisfyToQueryFilters(entityItem, filters),
+      );
+    }
+
+    if (orderBy) {
+      // todo this is absolutely hardcoded for now
+      // it makes assumption that we use exactly id prop for ordering
+      // and id is exactly `kind of number` so we can parse it.
+      const { column, direction } = orderBy[0];
+
+      entities.sort(
+        (item: TEntity, nextItem: TEntity): number =>
+          parseInt(item[column], 10) - parseInt(nextItem[column], 10),
+      );
+
+      if (direction === 'desc') {
```

Comment:
Why not pass this as a parameter to OData?  Just reversing the entities will result an an incorrect dataset.

Reply:
its both, in odata and in store...In the store I'm basically trying to parse queryOptions which I use inside list to make it consistent...So queryOptions is one source of truth as filter for fetching and gettings items from the store,  If I don't do this, I can't use it as cachable entity store, I'll have to clear it every time for every list/whatever.

Reply:
This doesn't make sense.  OData should always be the source of truth and however the data is stored should already match the query options.

Reply:
I also do not care about caching.  It seems like when you focus on this, it ends up making everything much more complicated.